silverback/types.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional, Protocol, Union
+
+from pydantic import BaseModel
+from typing_extensions import Self  # Introduced 3.11
+
+# NOTE: 'type | None' introduced in 3.10
+StrOrNone = Union[str, None]
+IntOrNone = Union[int, None]
```

Comment:
would prefer not using these, instead use `Optional[str]` for now (at least until we drop <3.10)

Reply:
They're not the same.  `Optional` implies its optionality as an argument.  A `Union` type requires the argument but allows a `None` value.

Reply:
Now I'm super confused how I got there.  I thought this came from a mypy error but things appear to be working fine without this union type.  Maybe leftover typescript behavior.  Will update, thanks for the docs link.

Reply:
> Maybe leftover typescript behavior.

to be fair, it's always appropiate to blame typescript